src/app/loadout-builder/process-worker/process.ts

```diff
@@ -398,7 +241,11 @@ export function process(
               continue;
             }
 
+            const armor = [helm, gaunt, chest, leg, classItem];
```

Comment:
Another optimisation could be to put off creating this array until we know we are going to keep the set.

Reply:
That's what this is - it's needed right away in `canTakeSlotIndependentMods`